### PR TITLE
update for mithril 2

### PR DESCRIPTION
**Pull request by askvortsov1**

### CODE REVIEW

**Review by clarkwinkelmann — COMMENTED** [review]
Since we're inside a class component and `drawChart` is bound to `this`, can't we just replace `vnode.state.chart` with `this.chart` ?

File: js/src/admin/components/StatisticsWidget.js [hunk 1]
```diff
@@ -110,12 +110,12 @@ export default class StatisticsWidget extends DashboardWidget {
...
       if (period.step < 86400) {
-        label = moment.unix(i + offset).utc().format('h A');
+        label = dayjs.unix(i + offset).format('h A');
       } else {
-        label = moment.unix(i + offset).utc().format('D MMM');
+        label = dayjs.unix(i + offset).format('D MMM');
 
         if (period.step > 86400) {
-          label += ' - ' + moment.unix(i + offset + period.step - 1).utc().format('D MMM');
+          label += ' - ' + dayjs.unix(i + offset + period.step - 1).utc().format('D MMM');
```
Comment (dsevillamartin) [inline js/src/admin/components/StatisticsWidget.js]: You didn't remove the `.utc()` call here.

I haven't checked anything else; I just noticed this from enabling it @ https://m2.flarum.site

**Review by dsevillamartin — CHANGES_REQUESTED** [review]

**Review by clarkwinkelmann — APPROVED** [review]
Seems good visually. I expect you have tested locally that may suggested changes are actually working? :innocent: 

I didn't know we could use `oncreate`/`onupdate` outside of components. We could also do those tasks methods on the Component's `oncreate`/`onupdate`. Right now I think it might be a bit confusing since `this` is the state of `StatisticsWidget` while `vnode` is a reference to the `.StatisticsWidget-chart` child element :thinking:

**Comment by askvortsov1** [comment]
I have. And iirc, mithril treats vnode with life cycle hooks as components?

**Comment by clarkwinkelmann** [comment]
If it works, I'm fine with the current approach. We can always revisit this at a later time if we decide to avoid this kind of lifecycle hooks.

**Comment by franzliedke** [comment]
Yeah, it would probably be less surprising when extracted to its own component / fragment. Let's add it to the long list of "nice to haves". :wink:

**Comment by franzliedke** [comment]
I extracted the moment -> dayjs changes to master.

**Review by franzliedke — APPROVED** [review]

**Pull request merged**